### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by chris-giblin**

### CODE REVIEW

File: 013-kafka-topic-encryption.md [hunk 1]
```diff
@@ -4,38 +4,93 @@
...
+message is forwarded to the broker. As a result, the topic data is stored by the broker in encrypted form.
+On the reverse direction, encrypted responses (responses to Fetch requests) are decrypted prior to being sent to clients.
+
+As defined in the encryption policy, each topic can be encrypted by a different key,
+allowing brokers to store a mix of encrypted and unencrypted data, where
+data owners can manage the keys to their topics.
+Keys ideally are stored in a key management system with access policies and logging.
+
+
+A core topic-encryption component, termed the _Encryption Module_, is proposed which is then deployed in a proxy. The
```
Comment (zrlvep) [inline 013-kafka-topic-encryption.md:48]: Typo. "the" at the end

File: 013-kafka-topic-encryption.md [hunk 2]
```diff
@@ -4,38 +4,93 @@
...
+
+
 
 ## Affected/not affected projects
 
 Call out the projects in the Strimzi organisation that are/are not affected by this proposal. 
-here comes text
+
 
 ## Compatibility
```
Comment (zrlvep) [inline 013-kafka-topic-encryption.md:98]: Should we add here that the proxy needs to be compatible to future message formats of the producerequest and fetchresponse?

File: 013-kafka-topic-encryption.md [hunk 3]
```diff
@@ -4,38 +4,93 @@
...
+
+### Proxy
+A proxy can be implemented as a free-standing process or as a sidecar. In both cases, the proxy needs to
+intercept and de-serialize the Kafka messages, encrypt/decrypt the relevant parts, 
+and reserialize them. The overhead of a proxy potentially has an effect on performance.
+It also means that the proxy version must always be in phase with that of the broker.
+A proxy-based solution however has the advantage that both Kafka client and broker are unaware
+of the proxy and do not require any modification or configuration to support encryption at rest.
+
+Initially a free-standing proxy will be developed.
```
Comment (zrlvep) [inline 013-kafka-topic-encryption.md]: Do we want to define this already or just state leave the two options?

**Review by zrlvep — APPROVED** [review]
A few comments. I think it looks good otherwise and should be presented to the Redhat folks.

**Review by SeanRooooney — COMMENTED** [review]
Very Meta Question.

Do we assume the proxy is a given here or is that part of the dicussion to be had ?

**Comment by chris-giblin** [comment]
Agree, we need to clarify this point for ourselves. I was not sure to what extent we want to indicate possibilities in this proposal or just tell it how it will be done.

The RH guys did not jump for joy over Envoy, with one person expressing, "the proxy does not necessarily have to be Envoy". Combining that with the expressed interest in both client-side and server-side proxies, I arrived at the standalone proxy being the first version. Envoy is a very interesting option, especially in Istio setting. Envoy also means we don't have to develop a proxy from scratch.

Perhaps we indicate readiness to discuss the prioritization.

**Comment by chris-giblin** [comment]
On second thought, an Envoy proxy can be deployed client side as well (mental mistake on my part).
Therefore I think Envoy should be the preferred implementation.

**Comment by SeanRooooney** [comment]
I think we should state:
(1) state this is a proxy based 
(2) outlined Rejected alternatives (seems others do this, in other proposals)
(3) callout envoy as a means (but not the only means) of implementing the proxy

**Review by SeanRooooney — APPROVED** [review]

**Comment by chris-giblin** [comment]
> state this is a proxy based

This is stated in the proposal introduction. Feel free to edit if you think it needs to be strengthened

**Comment by zrlvep** [comment]
> (2) outlined Rejected alternatives (seems others do this, in other proposals)

Should client and broker based options be listed here?

**Comment by chris-giblin** [comment]
> 2) outlined Rejected alternatives (seems others do this, in other proposals)
> (3) callout envoy as a means (but not the only means) of implementing the proxy

Done

**Pull request merged**